### PR TITLE
Add test step logging and screenshots to WebUI tests

### PR DESCRIPTION
**Pull request by velezd**
Adding decorator that prints the name of a function and its parameters (test step), and optionally prints docstring and creates screenshots. Also decorating all existing helper functions that interact with the WebUI.

From now on we should use the decorator for any new helper function that we create.

### CODE REVIEW

**Comment by velezd** [comment]
I'm not sure what do you mean by storing. The screenshots are created by this function [/common/testlib.py#L950](https://github.com/cockpit-project/cockpit/blob/41f8035eaf59d0a0ace6b86684bd8bd5c1444694/test/common/testlib.py#L950), and whatever runs the test has to collect them. If run by cockpit CI, they are stored for example [here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-4575-20230223-105157-3c18028c-fedora-rawhide-boot/index.html), if run by Permian it creates test report similar to what anabot did and saves it with other logs.

File: ui/webui/test/helpers/step_logger.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            if snapshots or snapshot_before:
+                BrowserSnapshot.new(args[0].browser)
+            
+            function(*args, **kwargs)
+            
+            if snapshots or snapshot_after:
+                BrowserSnapshot.new(args[0].browser)
+
+        return wrapper
+    return decorator
```
Comment (VladimirSlavik) [inline ui/webui/test/helpers/step_logger.py:46]: Might be nice to have one empty line here.

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

(It took me some looking to realize the screenshots are not assertions, but just logging.)

File: ui/webui/test/helpers/step_logger.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            if nice_args:
+                nice_args = ', with ' + nice_args
+
+            print(f'[TEST STEP] {function.__name__}{nice_args}')
+            if docstring:
+                print(f'[DOC] {function.__doc__}')
+            if snapshots or snapshot_before:
+                BrowserSnapshot.new(args[0].browser)
+            
+            function(*args, **kwargs)
```
Comment (marusak) [inline ui/webui/test/helpers/step_logger.py:40]: This function can have return value. Like for example [get_local_search](https://github.com/rhinstaller/anaconda/blob/master/ui/webui/test/helpers/language.py#L41) has. When you don't return it, python by default returns `None`. Which is why you see [here](https://github.com/rhinstaller/anaconda/pull/4575/files#diff-af148855d642cc07a588883e506315f3f74ff4f0ab87b832028ff650256ad626R69) `None`.

This simple fix makes the one failing test to pass:
```diff
--- ui/webui/test/helpers/step_logger.py
+++ ui/webui/test/helpers/step_logger.py
@@ -37,11 +37,13 @@ def log_step(snapshots=False, snapshot_before=False, snapshot_after=False, docst
             if snapshots or snapshot_before:
                 BrowserSnapshot.new(args[0].browser)
             
-            function(*args, **kwargs)
+            result = function(*args, **kwargs)
             
             if snapshots or snapshot_after:
                 BrowserSnapshot.new(args[0].browser)
 
+            return result
+
         return wrapper
     return decorator

```

Comment (velezd) [reply ui/webui/test/helpers/step_logger.py]: Good catch, thank you.

**Comment by VladimirSlavik** [comment]
@M4rtinK are you ok with merging this?

**Review by rvykydal — APPROVED** [review]
Looks good to me. Thank you.

**Pull request merged**